File: backend/backendTienda2/src/main/java/backend/rafhergom/tfg/model/entity/Producto.java
package backend.rafhergom.tfg.model.entity;

import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Date;

@Data
@Entity
@Table(name = "Producto")
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "nombre_producto", nullable = false, length = 100)
    private String nombreProducto;

    @Column(name = "descripcion", columnDefinition = "TEXT")
    private String descripcion;

    @Column(name = "precio", precision = 10, scale = 2, nullable = false)
    private BigDecimal precio;

    @Column(name = "stock_disponible", nullable = false)
    private int stockDisponible;

    @ManyToOne
    @JoinColumn(name = "categoria_producto_id")
    private CategoriaProducto categoriaProducto;
    
    @ManyToOne
    @JoinColumn(name = "negocio_id", nullable = false)
    private Negocio negocio;
    
    @Column(name = "imagenurl", nullable = false)
    private String imagenURL;
    @Column(name = "fecha_creacion", nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private Date fechaCreacion;

    @Column(name = "fecha_modificacion", nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP")
    private Date fechaModificacion;

    @Column(name = "usuario_creacion")
    private Long usuarioCreacion;

    @Column(name = "usuario_modificacion")
    private Long usuarioModificacion;
}

File: frontend/TiendaRafhergomFrontend/src/app/shared/model/dtos/producto.ts
export interface Producto {
  id: string;
  nombreProducto: string;
  descripcion: string;
  precio: string;
  stockDisponible: string;
  categoriaId: string;
}

export interface ProductoView {
  id: string;
  nombreProducto: string;
  descripcion: string;
  precio: string;
  stockDisponible: string;
  categoriaId: string;
  parentProducto: ProductoView;
}

export interface ProductoSimple {
  id: string;
  nombreProducto: string;
  descripcion: string;
  precio: string;
  stockDisponible: string;
  categoriaId: string;
}

